src/update_agent/actor.rs

```diff
@@ -76,10 +138,13 @@ impl Handler<RefreshTick> for UpdateAgent {
                     "scheduling next agent refresh in {} seconds",
                     pause.as_secs()
                 );
-                Self::tick_later(ctx, pause);
+                let handle = Self::tick_later(ctx, pause);
+                actor.tick_later_handle = Some(handle);
             } else {
                 let update_timestamp = chrono::Utc::now();
                 actor.state_changed = update_timestamp;
+                // Cancel scheduled ticks, if any.
+                actor.cancel_scheduled_ticks(&msg, ctx);
```

Comment:
I think here it is possible to have a race where a self tick msg and a command msg is sent at the same time, and the command msg arrvies first.
If this happens, we can't really cancel the scheduled tick like this because the message had alrady been sent. Though I don't think it is too big of a deal to have an extra tick; in practice, this results in two continuous ticks without any delay in between when Zincati is in a steady state (`NoNewUpdates` and `UpdateStaged`) where it periodically sends itself refresh ticks.